src/base/bittorrent/session.cpp

```diff
@@ -3068,6 +3070,8 @@ void Session::startUpTorrents()
         if (!addTorrent_impl(params.addTorrentData, params.magnetUri, TorrentInfo::loadFromFile(filePath), params.data))
             logger->addMessage(tr("Unable to resume torrent '%1'.", "e.g: Unable to resume torrent 'hash'.")
                                .arg(params.hash), Log::CRITICAL);
+        // don't waste memory with to many unprocessed queue entries
+        readAlerts();
```

Comment:
Doesn't this defeat the purpose of doing `session::async_add_torrent()`?
Also if I remember our workflow correctly, we run the risk of processing the alerts for the added torrents before the UI is ready to consume them.
Our `session` class is a singleton, and it instantiated before any GUI elements. But the processing of the alerts is done after we return control to the qt mainloop, but at that point all GUI elements are constructed and instantiated.